crates/sui-framework/packages/sui-system/sources/sui_system_state_inner.move

```diff
@@ -865,18 +865,29 @@ module sui_system::sui_system_state_inner {
 
         let storage_charge = storage_reward.value();
         let computation_charge = computation_reward.value();
-
+        let mut stake_subsidy = balance::zero();
         // Include stake subsidy in the rewards given out to validators and stakers.
         // Delay distributing any stake subsidies until after `stake_subsidy_start_epoch`.
         // And if this epoch is shorter than the regular epoch duration, don't distribute any stake subsidy.
-        let stake_subsidy =
-            if (ctx.epoch() >= self.parameters.stake_subsidy_start_epoch  &&
-                epoch_start_timestamp_ms >= prev_epoch_start_timestamp + self.parameters.epoch_duration_ms)
-            {
-                self.stake_subsidy.advance_epoch()
-            } else {
-                balance::zero()
+        if (ctx.epoch() >= self.parameters.stake_subsidy_start_epoch  &&
+            epoch_start_timestamp_ms >= prev_epoch_start_timestamp + self.parameters.epoch_duration_ms)
+        {
+            // special case for epoch 560 -> 561 change bug. add extra subsidies for "safe mode"
+            // where reward distribution was skipped. use distribution counter to avoiding affecting devnet
+            // and testnet
+            if (self.stake_subsidy.get_distribution_counter() == 540) {
+                let first_safe_mode_epoch = 561;
+                let mut i = 0;
+                let safe_mode_epoch_count = ctx.epoch() - first_safe_mode_epoch;
```

Comment:
We will fail with arithmetic error on testnet where `stake_subsidy_start_epoch=0`. When `distribution_counter` reaches `540`, the epoch will be `540`. As a way to prevent it, we could add an extra condition that distribution counter is smaller than the current epoch. But it doesn't seem right.

Reply:
Addressed this via your fix in the patch (add ctx.epoch > 560 above

Reply:
This still doesn't seem right.
561 is the first epoch that we enter safe mode, which means in that epoch we did not distribute subsidy.
ctx.epoch() is the epoch we are about to end, which is the last safe mode epoch.
So to make sure we distribute all subsidy it seems this should be `ctx.epoch() - first_safe_mode_epoch + 1`.
Am I thinking about it correctly?

Reply:
@lxfind I struggled with it today too. Distribution is still bumped once even if this condition is reached.

Reply:
Oh I see there is another one afterwards. That makes sense. Thanks @damirka!

Reply:
Yes, I wrote the code to fall through to the normal case. That was maybe not so clear.